### PR TITLE
${level} add format option (full, single char and ordinal)

### PR DESCRIPTION
**Pull request by c0shea**
Adds single character and ordinal level output options to the LevelLayoutRenderer to address #1949 and #1950

### CODE REVIEW

**Comment by 304NotModified** [comment]
cool! Thanks!

Could you place the `enum` in a new file and outside the class? Please check other enums for the naming style. 

to sync the csproj files, see https://github.com/NLog/NLog/blob/master/CONTRIBUTING.md#sync-projects

File: src/NLog/LayoutRenderers/LevelLayoutRenderer.cs [hunk 1]
```diff
@@ -46,13 +48,44 @@ namespace NLog.LayoutRenderers
     public class LevelLayoutRenderer : LayoutRenderer
     {
         /// <summary>
+        /// A value indicating when the cache is cleared.
```
Comment (304NotModified) [inline src/NLog/LayoutRenderers/LevelLayoutRenderer.cs]: I think this is wrong docs?

File: src/NLog/LayoutRenderers/LevelLayoutRenderer.cs [hunk 2]
```diff
@@ -46,13 +48,44 @@ namespace NLog.LayoutRenderers
...
+        /// </summary>
+        [Flags]
+        public enum LevelOutputOption
+        {
+            /// <summary>Ouput the name as the level.</summary>
```
Comment (304NotModified) [inline src/NLog/LayoutRenderers/LevelLayoutRenderer.cs]: Proposal: 
- render the full level name
- render the first character for the level
- render the ordinal (aka number) for the level.

File: src/NLog/LayoutRenderers/LevelLayoutRenderer.cs [hunk 3]
```diff
@@ -46,13 +48,44 @@ namespace NLog.LayoutRenderers
...
+            SingleCharacter = 1,
+            /// <summary>Output an ordinal (i.e. number) as the level.</summary>
+            Ordinal = 2
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating how the level is output.
+        /// </summary>
+        [DefaultValue(LevelOutputOption.Name)]
+        public LevelOutputOption LevelOutput { get; set; }
```
Comment (304NotModified) [inline src/NLog/LayoutRenderers/LevelLayoutRenderer.cs]: I was thinking of "format", what do you think?

File: tests/NLog.UnitTests/LayoutRenderers/LogLevelTests.cs [hunk 4]
```diff
@@ -60,5 +60,53 @@ public void LogLevelTest()
...
+            logger.Debug("a");
+            AssertDebugLastMessage("debug", "D a");
+            logger.Info("a");
+            AssertDebugLastMessage("debug", "I a");
+            logger.Warn("a");
+            AssertDebugLastMessage("debug", "W a");
+            logger.Error("a");
+            AssertDebugLastMessage("debug", "E a");
+            logger.Fatal("a");
+            AssertDebugLastMessage("debug", "F a");
```
Comment (304NotModified) [inline tests/NLog.UnitTests/LayoutRenderers/LogLevelTests.cs]: looks good! please add "trace"

File: tests/NLog.UnitTests/LayoutRenderers/LogLevelTests.cs [hunk 5]
```diff
@@ -60,5 +60,53 @@ public void LogLevelTest()
...
+            logger.Debug("a");
+            AssertDebugLastMessage("debug", "1 a");
+            logger.Info("a");
+            AssertDebugLastMessage("debug", "2 a");
+            logger.Warn("a");
+            AssertDebugLastMessage("debug", "3 a");
+            logger.Error("a");
+            AssertDebugLastMessage("debug", "4 a");
+            logger.Fatal("a");
+            AssertDebugLastMessage("debug", "5 a");
```
Comment (304NotModified) [inline tests/NLog.UnitTests/LayoutRenderers/LogLevelTests.cs]: also please add trace

**Comment by c0shea** [comment]
You're welcome :)

I like your suggestion to name it Format, so I renamed the enum to LevelFormat and moved it to its own file. I modeled it after StackTraceFormat, so hopefully it follows the style. I also removed the wrong doc and updated the unit tests to add trace.

Let me know if you need anything else to accept.

**Comment by 304NotModified** [comment]
Looks ready to merge! Could you please update the docs? Thanks!

**Comment by 304NotModified** [comment]
Fixes #1949 and fixes #1950

**Comment by 304NotModified** [comment]
docs see https://github.com/NLog/NLog/issues/2043

Will merge this an release a beta

**Comment by c0shea** [comment]
Thanks! I have updated the docs with the new parameter and added a condensed table.

**Comment by 304NotModified** [comment]
FYI,

NLog 4.4.6 is live! 

https://www.nuget.org/packages/NLog/4.4.6

**Pull request merged**